### PR TITLE
SDU-1875 introduce zwave specific service

### PR DESCRIPTION
**Pull request by eternalamenity**
If someone wants to know more about why this service is being created, PM me or write a comment.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1243,3 +1243,78 @@ out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade s
...
+
+#### Service name 
+
+`zwave_specific`
+
+#### Interfaces
+
+Type | Interface               | Value type | Description 
+-----|-------------------------|------------|--------------
+out  | evt.notification.report | str_map    | Notification Reports that are sent by slave devices.
```
Comment (bjhandeland) [inline README.md:1258]: nit: change to `Notification reports that are sent by worker devices.` 

I changed slave to worker, but if slave is used in the zwave documentation, we should stick with that.

File: README.md [hunk 2]
```diff
@@ -1243,3 +1243,78 @@ out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade s
...
+
+#### Storage property
+
+Key        | Value description 
+-----------|----------------------------------
+sub_value  | Values pair to store in Vinculum.
```
Comment (bjhandeland) [inline README.md:1264]: I don't get what `sub_value` is? Subscription value? Typo for supported value? And I don't get what the description means either :|

File: README.md [hunk 3]
```diff
@@ -1243,3 +1243,78 @@ out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade s
...
+
+#### Examples
+
+Motion Detection occurs for state Motion Sensor Status:
```
Comment (bjhandeland) [inline README.md:1268]: I would use the value names in the header here to make it easier to read the example, e.g.

"`motion_detection` detected for state `motion_sensor_status`.".

File: README.md [hunk 4]
```diff
@@ -1243,3 +1243,78 @@ out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade s
...
+
+```json
+{
+   "type": "evt.notification.report",
+   "serv": "zwave_specific",
+   "val_t": "str_map",
+   "val": {
+      "notificationEventState": "motion_detection",
+      "notificationType": "home_security",
+      "stateVariable": "motion_sensor_status"
```
Comment (bjhandeland) [inline README.md:1278]: If these are zwave specific key names, can you please dcoument what they represent? It's not immediately clear to me what they are.

If they are not zwave specific and they are keys we have added, can we revise the names?

notificationType: type of the notification, gotcha.
notificationEventState: why not just notificationEvent?
stateVariable: why not just notificationState

Either way, these are not that clear and should be documented / revised.

File: README.md [hunk 5]
```diff
@@ -1243,3 +1243,78 @@ out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade s
...
+   },
+   "storage": {
+     "sub_value": "home_security:motion_sensor_status"
+   }
+}
+```
+
+Motion Sensor Status state has been changed to State Idle:
```
Comment (bjhandeland) [inline README.md:1286]: Change title to use value names for clarity.

**Review by bjhandeland — CHANGES_REQUESTED** [review]
see comments

File: README.md (hunk 2 above)
Comment (eternalamenity) [reply README.md:1264]: It defines how values are stored in Vinculum persistant storage. It defines a pair of keys which should be treated as unique to hold as persistant data. 
For example now for every evt.notification.report Vinculum would keep only one value.
But I want a Vinculum to hold every value with unique pair of keys eg. "home_security:motion_sensor_status".
So in that way when another notification occurs (eg. home_security:cover_status), Vinculum will not replace previous one, but will keep both.

File: README.md (hunk 1 above)
Comment (eternalamenity) [reply README.md:1258]: From zwave perspective "slave" means a device that is added to a so-called "controller" and cannot operate itself with network's capabilities. So I think we can stick with that.

Comment (bjhandeland) [reply README.md:1258]: Sounds good. Still change it to `Notification reports that are sent by slave devices.` :)

**Review by bjhandeland — APPROVED** [review]
lgtm

File: README.md (hunk 4 above)
Comment (eternalamenity) [reply README.md]: fixed, use proper description now

**Review by bjhandeland — APPROVED** [review]
lgtm

**Pull request merged**